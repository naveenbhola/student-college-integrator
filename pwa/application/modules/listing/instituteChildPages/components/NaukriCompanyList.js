import PropTypes from 'prop-types'
import React, {Component} from 'react';
import './../assets/naukriCompanyList.css';
import {getObjectSize} from './../../../../utils/commonHelper';
import Analytics from './../../../reusable/utils/AnalyticsTracking';
import Lazyload from './../../../reusable/components/Lazyload';

class NaukriCompanyList extends Component {
    constructor(props) {
        super(props);
        this.state = {
            numberOfCompanyList :10
        }
    }

    trackEvent(actionLabel,label)
    {
      var category = "Placement_Page";
      if(this.props.gaTrackingCategory){
        category = this.props.gaTrackingCategory; 
      }
      Analytics.event({category : category, action : actionLabel, label : label});
    }


    companyList  =  () => {
        const {companyData}  = this.props;
        let number = getObjectSize(companyData);
        let maxSalary = 0;
         maxSalary = companyData[0].avgSalary;
        let companyList = [];
        let index =0;
        for(let i in companyData){
            index++;
            if(index > this.state.numberOfCompanyList){
                companyList.push(<label key={'comany_'+index} onClick={() => this.viewAllCompany()} id='view_all_company' className="arrow">{'View All '+number}</label>);
                break;
            }
            let salary =  parseInt((companyData[i].avgSalary/maxSalary)*100);
            let pageHtml =
                (
                    <div className={"organizer_list"} key={'comany1_'+index}>
                        <div className="organizer_logo">
                            <Lazyload offset={100} once>
                                <img src={companyData[i].companyLogoFileName} alt="sample image" />
                            </Lazyload>
                        </div>
                        <div className="organizer_dtls">
                            <p className="organizer_name">{companyData[i].companyName}</p>
                            <div className="progress_block">
                                <div className="avg_package">Avg &#8377; {companyData[i].avgSalary}L <span>|</span> {companyData[i].numAlumni} Alumni</div>
                                <div className="packagebar">
                                    <div className="avg_packagebar" style={{width: salary + '%'}}></div>
                                </div>
                            </div>
                        </div>
                    </div>
                );
            companyList.push(pageHtml);
        }
        return companyList;
    }

    viewAllCompany(){
        this.trackEvent("Click","view_all_20_companies_Naukri_placement")
        this.setState({'numberOfCompanyList':25});
    }


    render(){
        const {companyData,totalCompanies}  = this.props;
        if(totalCompanies ==0){
            return null;
        }   
        return(
            <div className="salary_stats" id='companyList'>
                <strong className="organzr_titl">How much they earn</strong>
                <div className='salary_stats_desk'>
                    {this.companyList()}

                </div>
            </div>
        )
    }

}

export default NaukriCompanyList;

NaukriCompanyList.propTypes = {
    companyData: PropTypes.any
}