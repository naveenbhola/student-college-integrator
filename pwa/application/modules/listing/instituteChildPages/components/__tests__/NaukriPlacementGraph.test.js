import React from 'react';
import { shallow,mount } from 'enzyme';
import NaukriCompanyList from  "./../NaukriPlacementGraph";
import {MockPlacementData} from "./../PlacementMockData";

const setup = propOverrides => {
	const props = Object.assign({
		salaryData : MockPlacementData.naukriData.salaryData

	},propOverrides)	


  const wrapper = shallow(<NaukriCompanyList {...props} />);
  return {
    props,
    wrapper
  }
}


describe('Naukri graph with All Data present', () => {
  test('snapshot test' , () => {
  	const {wrapper} = setup({});
  	expect(wrapper).toMatchSnapshot();
  })

  test('minimum salary null' , () => {
    let salaryData = Object.assign({},MockPlacementData.naukriData.salaryData);
    salaryData.minSalary = null;

    const {wrapper} = setup({salaryData});
    expect(wrapper.state().data.length).toEqual(3);
  })
  test('avg salary null' , () => {
    let salaryData = Object.assign({},MockPlacementData.naukriData.salaryData);
    salaryData.avgSalary = null;

    const {wrapper} = setup({salaryData});
    expect(wrapper.state().data.length).toEqual(3);
  })
  test('max salary null' , () => {
    let salaryData = Object.assign({},MockPlacementData.naukriData.salaryData);
    salaryData.maxSalary = null;

    const {wrapper} = setup({salaryData});
    expect(wrapper.state().data.length).toEqual(3);
  })
  test('min and avg both salary null' , () => {
    let salaryData = Object.assign({},MockPlacementData.naukriData.salaryData);
    salaryData.minSalary = null;
    salaryData.avgSalary = null;

    const {wrapper} = setup({salaryData});
    expect(wrapper.state().data.length).toEqual(2);
  })
  test('min and avg and max all salary null' , () => {
    let salaryData = Object.assign({},MockPlacementData.naukriData.salaryData);
    salaryData.minSalary = null;
    salaryData.avgSalary = null;
    salaryData.maxSalary = null;
    const {wrapper} = setup({salaryData});
    expect(wrapper.state().data.length).toEqual(1);
  })

});
