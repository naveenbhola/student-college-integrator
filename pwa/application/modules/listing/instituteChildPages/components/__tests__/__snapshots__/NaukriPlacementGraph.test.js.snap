// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Naukri graph with All Data present snapshot test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NaukriPlacementGraph
    salaryData={
      Object {
        "avgSalary": 14.35,
        "maxSalary": 37,
        "minSalary": 8,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <n
        allowEmptyRows={false}
        chartActions={null}
        chartEvents={Array []}
        chartLanguage="en"
        chartPackages={
          Array [
            "corechart",
          ]
        }
        chartType="ColumnChart"
        chartVersion="current"
        columns={Array []}
        data={
          Array [
            Array [
              "",
              "",
              Object {
                "role": "style",
              },
              Object {
                "role": "annotation",
              },
              Object {
                "role": "tooltip",
              },
            ],
            Array [
              "Min",
              8,
              "#008489",
              "8 lakh",
              "8 lakh",
            ],
            Array [
              "Median",
              14.35,
              "#008489",
              "14.35 lakh",
              "14.35 lakh",
            ],
            Array [
              "Max",
              37,
              "#008489",
              "37 lakh",
              "37 lakh",
            ],
          ]
        }
        dateFormat={null}
        diffdata={null}
        formatters={Array []}
        graph_id="LineChart2"
        height="250px"
        legend_toggle={false}
        loadCharts={true}
        loader={
          <div>
            Rendering Chart
          </div>
        }
        numberFormat={null}
        options={
          Object {
            "annotations": Object {
              "alwaysOutside": true,
              "textStyle": Object {
                "color": "black",
                "fontSize": 11,
              },
            },
            "bar": Object {
              "groupWidth": "50%",
            },
            "chartArea": Object {
              "width": "80%",
            },
            "legend": Object {
              "position": "none",
            },
            "seriesType": "bars",
            "tooltip": Object {
              "textStyle": Object {
                "fontName": "Verdana",
                "fontSize": 12,
                "fontWeight": 600,
              },
            },
            "vAxis": Object {
              "minValue": 0,
              "title": "Salary in (lakh)",
            },
          }
        }
        rows={Array []}
        width="100%"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allowEmptyRows": false,
        "chartActions": null,
        "chartEvents": Array [],
        "chartLanguage": "en",
        "chartPackages": Array [
          "corechart",
        ],
        "chartType": "ColumnChart",
        "chartVersion": "current",
        "columns": Array [],
        "data": Array [
          Array [
            "",
            "",
            Object {
              "role": "style",
            },
            Object {
              "role": "annotation",
            },
            Object {
              "role": "tooltip",
            },
          ],
          Array [
            "Min",
            8,
            "#008489",
            "8 lakh",
            "8 lakh",
          ],
          Array [
            "Median",
            14.35,
            "#008489",
            "14.35 lakh",
            "14.35 lakh",
          ],
          Array [
            "Max",
            37,
            "#008489",
            "37 lakh",
            "37 lakh",
          ],
        ],
        "dateFormat": null,
        "diffdata": null,
        "formatters": Array [],
        "graph_id": "LineChart2",
        "height": "250px",
        "legend_toggle": false,
        "loadCharts": true,
        "loader": <div>
          Rendering Chart
        </div>,
        "numberFormat": null,
        "options": Object {
          "annotations": Object {
            "alwaysOutside": true,
            "textStyle": Object {
              "color": "black",
              "fontSize": 11,
            },
          },
          "bar": Object {
            "groupWidth": "50%",
          },
          "chartArea": Object {
            "width": "80%",
          },
          "legend": Object {
            "position": "none",
          },
          "seriesType": "bars",
          "tooltip": Object {
            "textStyle": Object {
              "fontName": "Verdana",
              "fontSize": 12,
              "fontWeight": 600,
            },
          },
          "vAxis": Object {
            "minValue": 0,
            "title": "Salary in (lakh)",
          },
        },
        "rows": Array [],
        "width": "100%",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <n
          allowEmptyRows={false}
          chartActions={null}
          chartEvents={Array []}
          chartLanguage="en"
          chartPackages={
            Array [
              "corechart",
            ]
          }
          chartType="ColumnChart"
          chartVersion="current"
          columns={Array []}
          data={
            Array [
              Array [
                "",
                "",
                Object {
                  "role": "style",
                },
                Object {
                  "role": "annotation",
                },
                Object {
                  "role": "tooltip",
                },
              ],
              Array [
                "Min",
                8,
                "#008489",
                "8 lakh",
                "8 lakh",
              ],
              Array [
                "Median",
                14.35,
                "#008489",
                "14.35 lakh",
                "14.35 lakh",
              ],
              Array [
                "Max",
                37,
                "#008489",
                "37 lakh",
                "37 lakh",
              ],
            ]
          }
          dateFormat={null}
          diffdata={null}
          formatters={Array []}
          graph_id="LineChart2"
          height="250px"
          legend_toggle={false}
          loadCharts={true}
          loader={
            <div>
              Rendering Chart
            </div>
          }
          numberFormat={null}
          options={
            Object {
              "annotations": Object {
                "alwaysOutside": true,
                "textStyle": Object {
                  "color": "black",
                  "fontSize": 11,
                },
              },
              "bar": Object {
                "groupWidth": "50%",
              },
              "chartArea": Object {
                "width": "80%",
              },
              "legend": Object {
                "position": "none",
              },
              "seriesType": "bars",
              "tooltip": Object {
                "textStyle": Object {
                  "fontName": "Verdana",
                  "fontSize": 12,
                  "fontWeight": 600,
                },
              },
              "vAxis": Object {
                "minValue": 0,
                "title": "Salary in (lakh)",
              },
            }
          }
          rows={Array []}
          width="100%"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowEmptyRows": false,
          "chartActions": null,
          "chartEvents": Array [],
          "chartLanguage": "en",
          "chartPackages": Array [
            "corechart",
          ],
          "chartType": "ColumnChart",
          "chartVersion": "current",
          "columns": Array [],
          "data": Array [
            Array [
              "",
              "",
              Object {
                "role": "style",
              },
              Object {
                "role": "annotation",
              },
              Object {
                "role": "tooltip",
              },
            ],
            Array [
              "Min",
              8,
              "#008489",
              "8 lakh",
              "8 lakh",
            ],
            Array [
              "Median",
              14.35,
              "#008489",
              "14.35 lakh",
              "14.35 lakh",
            ],
            Array [
              "Max",
              37,
              "#008489",
              "37 lakh",
              "37 lakh",
            ],
          ],
          "dateFormat": null,
          "diffdata": null,
          "formatters": Array [],
          "graph_id": "LineChart2",
          "height": "250px",
          "legend_toggle": false,
          "loadCharts": true,
          "loader": <div>
            Rendering Chart
          </div>,
          "numberFormat": null,
          "options": Object {
            "annotations": Object {
              "alwaysOutside": true,
              "textStyle": Object {
                "color": "black",
                "fontSize": 11,
              },
            },
            "bar": Object {
              "groupWidth": "50%",
            },
            "chartArea": Object {
              "width": "80%",
            },
            "legend": Object {
              "position": "none",
            },
            "seriesType": "bars",
            "tooltip": Object {
              "textStyle": Object {
                "fontName": "Verdana",
                "fontSize": 12,
                "fontWeight": 600,
              },
            },
            "vAxis": Object {
              "minValue": 0,
              "title": "Salary in (lakh)",
            },
          },
          "rows": Array [],
          "width": "100%",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
